From c7f8bac5dadc44ba8ad8674b3023bd901030e553 Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Wed, 20 Mar 2024 10:28:43 +0000
Subject: [PATCH] rdk/grr: i3c_master_unregister() has been changed to void

i3c_master_unregister() was changed to be void in kernel commit
0f74f8b6675c ("i3c: Make i3c_master_unregister() return void").

Upstream-Status: Inappropriate [RDK is released in tarball only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 klm/i3c/i3c-rdk.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/klm/i3c/i3c-rdk.c b/klm/i3c/i3c-rdk.c
index aaf429d..2818538 100644
--- a/klm/i3c/i3c-rdk.c
+++ b/klm/i3c/i3c-rdk.c
@@ -2677,7 +2677,6 @@ static int i3c_rdk_remove(struct platform_device *pdev)
 	struct resource *res = i3c_rdk_device->res;
 	void __iomem *regs = i3c_rdk_device->regs;
 	struct device *dev = &pdev->dev;
-	int rc;
 
 #ifdef CONFIG_DEBUG_FS
 	i3c_rdk_exit_debugfs(i3c_rdk_device);
@@ -2689,10 +2688,6 @@ static int i3c_rdk_remove(struct platform_device *pdev)
 
 	devm_iounmap(dev, regs);
 	devm_release_mem_region(dev, res->start, resource_size(res));
-	rc = i3c_master_unregister(&i3c_rdk_device->base);
-
-	if (rc)
-		return rc;
 
 	class_destroy(i3c_rdk_device->class);
 	unregister_chrdev_region(i3c_rdk_device->dev_t, I3C_RDK_MAX_DEVICES);
-- 
2.39.3

